ssas/service/admin/api.go

```diff
@@ -754,6 +754,61 @@ func deleteToken(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusAccepted)
 }
 
+func createKey(w http.ResponseWriter, r *http.Request) {
+	systemID := chi.URLParam(r, "systemID")
+	trackingID := ssas.RandomHexID()
+	keyEvent := ssas.Event{Op: "CreateKey", TrackingID: trackingID, Help: "calling from admin.createKey()"}
+
+	system, err := ssas.GetSystemByID(systemID)
+	if err != nil {
+		ssas.OperationFailed(keyEvent)
+		jsonError(w, http.StatusNotFound, "Invalid system ID")
+		return
+	}
+
+	var pk ssas.PublicKeyInput
+	if err := json.NewDecoder(r.Body).Decode(&pk); err != nil {
+		ssas.OperationFailed(keyEvent)
+		jsonError(w, http.StatusBadRequest, "Failed to read body")
+		return
+	}
+
+	if err := system.AddAdditionalPublicKey(strings.NewReader(pk.PublicKey), pk.Signature); err != nil {
+		ssas.OperationFailed(keyEvent)
+		jsonError(w, http.StatusInternalServerError, "Failed to add additional public key")
+		return
+	}
+
+	key, _ := system.GetEncryptionKey(trackingID)
+
+	w.Header().Set("Content-Type", "application/json")
+	keyStr := strings.Replace(key.Body, "\n", "\\n", -1)
+	fmt.Fprintf(w, `{ "client_id": "%s", "public_key": "%s" }`, system.ClientID, keyStr)
```

Comment:
We won't need it (at least right away, so I won't insist), but it's a good practice to output the ID of the object that was created. In `deleteKey()` the user needs it.